pkg/deployment/deployment_item.go

```diff
@@ -47,6 +48,55 @@ type DeploymentItem struct {
 	renderedYamlPath      string
 }
 
+// TODO: find better name then DeploymentItem...Vars?
+// IDEA: Make these sub-objects of actual di/dp
+type DeploymentItemProjectVars struct { // info about the deployment.yaml project that included this item
+	Parent  *DeploymentItemProjectVars // <parent project info, same structure as `project`>
+	DirName string                     `yaml:"dirName" validate:"required"` // dirname
+	RelPath []string                   // <relative directory to current source>
+	//RelRoot string                     // rel path to the project root dir, ie: ../../..
+	Tags *utils.OrderedMap // all tags, including the ones merged from the parents
+}
+type DeploymentItemVars struct {
+	DirName string                      `yaml:"dirName" validate:"required"` // dirname
+	DirPath string                      `yaml:"dirPath,omitempty"`           // full path to deploymentItem dir
```

Comment:
omitempty should probably be avoided in this case as it would lead to a lot of Jinja2 errors when people try to use non-existing/undefined values.